zcash_primitives/src/legacy.rs

```diff
@@ -407,6 +408,24 @@ impl TransparentAddress {
     }
 }
 
+impl TryFromAddress for TransparentAddress {
```

Comment:
This should implement `TryFromRawAddress`, as `TransparentAddress` is a network-agnostic type.